server/src/test/java/org/elasticsearch/indices/recovery/RecoveryTests.java

```diff
@@ -412,7 +412,9 @@ public void testShouldFlushAfterPeerRecovery() throws Exception {
             replica.indexSettings().updateIndexMetaData(builder.build());
             replica.onSettingsChanged();
             shards.recoverReplica(replica);
-            // Make sure the flushing will eventually be completed (eg. `shouldPeriodicallyFlush` is false)
+            // trigger a flush if `shouldPeriodicallyFlush` becomes true after we roll a new generation in RecoveryTarget#finalizeRecovery
+            replica.afterWriteOperation();
```

Comment:
shouldn't we call `afterWriteOperation` in `finalizeRecovery`?

Reply:
I preferred not to call `afterWriteOperation` in `finalizeRecovery` because rolling a new translog generation is not really a write. I pushed https://github.com/elastic/elasticsearch/pull/45182/commits/14d74b41e69b19a0d1490ba0d0c94c056a2460f5.